app/src/main/java/de/danoeh/antennapod/dialog/SubscriptionFilterDialog.java

```diff
@@ -0,0 +1,137 @@
+package de.danoeh.antennapod.dialog;
+
+import android.app.Dialog;
+import android.content.Context;
+
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.Spinner;
+
+import java.util.ArrayList;
+
+import de.danoeh.antennapod.R;
+
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_ALWAYS;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_AUTO_DELETE;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_AUTO_DOWNLOAD;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_DISABLE;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_ENABLE;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_GLOBAL_DEFAULT;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_KEEP_UPDATED;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_NEVER;
+import static de.danoeh.antennapod.fragment.SubscriptionFragment.STRING_NO_FILTER;
+
+public abstract class SubscriptionFilterDialog {
+
+    private final Context context;
+    final Dialog dialog;
+    private Spinner spinnerAutoDownload;
+    private Spinner spinnerKeepUpdated;
+    private Spinner spinnerAutoDelete;
+    private Button buttonConfirm;
+    private Button buttonCancel;
+    private ArrayList<String> spinnerArrayAutoDown_keepUpdated;
+    private ArrayList<String> spinnerArrayAutoDelete;
+    private String autoDownload;
+    private String keepUpdated;
+    private String autoDelete;
+
+    protected SubscriptionFilterDialog(Context theContext,String theAutoDownload, String theKeepUpdated,
+                                       String theAutoDelete) {
+        this.context = theContext;
+        this.autoDownload = theAutoDownload;
+        this.keepUpdated = theKeepUpdated;
+        this.autoDelete = theAutoDelete;
+
+        dialog = new Dialog(context);
```

Comment:
Please use an `AlertDialog.Builder`. This allows to set the title in a consistent style. Also, it allows to add buttons easily with `setPositiveButton` and `setNegativeButton`